ProcessLib/BoundaryCondition/GenericNaturalBoundaryCondition-impl.h

```diff
@@ -55,8 +55,8 @@ GenericNaturalBoundaryCondition<BoundaryConditionData,
     auto const& mesh_subset =
         dof_table_bulk.getMeshSubset(variable_id, component_id);
 
-    MeshLib::MeshSubset bc_mesh_subset =
-        mesh_subset.getIntersectionByNodes(nodes);
+    _nodes_subset = nodesNodesIntersection(mesh_subset.getNodes(), nodes);
+    MeshLib::MeshSubset bc_mesh_subset(mesh_subset.getMesh(), &_nodes_subset);
```

Comment:
Wouldn't it be better to store the nodes set as a `shared_ptr` inside the subset instead of stroing it in the caller?